### DISCUSSION TITLE
Setting up Fabric - Pbpaste works, but issues with YT and default model

### DISCUSSION THREAD

**Discussion by wanaka5** [danielmiessler/Fabric#1175] [Q&A]
Hi, I am non-technical, but attempting to set up fabric on my Mac. I saw some old tutorial videos, but those appear to be out of date with the transition from Python to Go. Good news is that I've set up Fabric and can prompt using pbpaste (e.g., copying text from a website `pbpaste | fabric -m llama3:latest -sp extract_wisdom`). However, I run into issues with the following:

1. Default Model: when using pbpaste, I have to specify -m llama3:latest every time. If I don't, it shows:

'Requested Model = 
 Default Model = 
 Default Vendor = .

Please run, fabric --setup, and select default model and vendor.'

I've set llama3 to default when using 'fabric -d', but this doesn't appear to fix anything. I've seen some discussion forums wrt issues w llama run locally, but mine works fine, it just needs to be in the command.

2. Youtube transcript: I've tried running two different commands to pull YT transcripts, one without quotations around the URL and one (this was the fix in another discussion) with. I get two different error messages.

`yt --transcript https://www.youtube.com/watch?v=UbDyjIIGaxQ | fabric -m llama3:latest -sp extract_wisdom`
- I get the following: "zsh: no matches found: https://www.youtube.com/watch?v=UbDyjIIGaxQ"

If I run with quotations:
`yt --transcript "https://www.youtube.com/watch?v=UbDyjIIGaxQ" | fabric -m llama3:latest -sp extract_wisdom`
- I get: "zsh: command not found: yt"

In both instances I get an output from fabric saying "I'm ready to extract valuable insights from the input. Please provide the text content..." Note that I have tried both of the above with '-y' instead of '-yt' also I have added my YT API key to fabric and set it up in my environment via `export YOUTUBE_API_KEY="YOUR_API_KEY"`

Apologies if these appear obvious, I've tried to look through the existing discussions but keep running into some minor snags. Any help would be greatly appreciated. Thanks.

**Comment by wanaka5** [comment]
I have solved the issues related to pulling the YT transcript. The updated code is `fabric -y "[insert YouTube URL]" | fabric -m [insert model] -sp [insert pattern]`

Since I am learning the ropes, I am using Ollama to use an open source model, but have also plugged in a Claude API. The difference between Llama3 and Sonnet 3.5 is night and day in terms of output. The Sonnet model crushes the small local version of llama I have so FYI for any newbies like me wondering why the outputs are really poor quality.

I have not found a solve on resolving the default model, despite having it selected through the setup.
<img width="434" alt="image" src="https://github.com/user-attachments/assets/669ff486-5b16-4569-8be0-94e19c54f2fa">

**Comment by JK95L** [comment]
wanaka5 solved the issue for me:
"I have solved the issues related to pulling the YT transcript. The updated code is fabric -y "[insert YouTube URL]" | fabric -m [insert model] -sp [insert pattern]"

**Comment by jmd1010** [comment]
And if you decide to use the web interface,  no need to enter qualifiers, simply paste the youtube url.
https://github.com/danielmiessler/fabric/blob/main/Web%20Interface%20MOD%20Readme%20Files/WEB%20V2%20Install%20Guide.md

**Comment by akiemw** [comment]
<img width="1049" alt="Screenshot 2025-04-04 at 3 19 02 PM" src="https://github.com/user-attachments/assets/095873d6-1ba6-4456-a87c-b746e91fd75e" />
Just doing a fresh install and not sure if that usually string is what is actually used to do my api call to Ollama. But that has never been my usual Ollama URL so that may be a problem. I want to test it but if someone knows off hand whether skipping inputting the actual usual Ollama address causes it to use the default as printed or if it checks for the global variable.

According to ollama's github docs as of today's date it's `http://localhost:11434/` 

(update) 
Used that exact url and I'm updated and running on fabric. So just a suggestion correcting to that usually prompt.
